### ISSUE TITLE
Notebook files (.snb) should be pretty printed when downloading them through the File menu

### ISSUE THREAD

**Issue by benqua** [andypetrella/spark-notebook#558]
When a notebook is exported through the "File -> Download as -> Spark Notebook" menu, it is not pretty printed. Everything is on a single line as shown by the unix file command:
$ file explore.snb 
explore.snb: HTML document, ASCII text, with very long lines, with no line terminators

**Comment by vidma** [comment]
see not much difference. it is not quite human readable and maybe be 1MB or more (in future) in size.

**Comment by andypetrella** [comment]
@vidma right, but there is also the willing here to use these downloaded files in Git for instance, hence the changes are easier to process

**Comment by benqua** [comment]
Right, I am just looking for away to version my notebook with git (or any line based versioning tool).

The snb format is not very good for this, because it also contains the result of the evaluation of the cell.
The .scala files would be good, but they don't contains the markdown cells.

A good fit for this use case would be the .scala file with the markdown cells added in comment.
Of course, this would only work if it is then possible to import back those files.
